plugins/CoreUpdater/Updater.php

```diff
@@ -131,7 +131,9 @@ public function updatePiwik($https = true)
         $response = Http::sendHttpRequest($partTwoUrl, 300);
         $response = @json_decode($response, $assoc = true);
 
-        $messages = array_merge($messages, $response);
+        if (!empty($response)) {
+            $messages = array_merge($messages, $response);
+        }
```

Comment:
In case there is an exception, should we throw an UpdaterException? Eg, we could in `oneClickUpdatePartTwo()` return `{"error": true, "messages": [...]}` and throw the exception as it would have been before.

Reply:
Might be good to add a quick test for this too, like a unit test for the controller that uses a mock updater. Not sure if it's worth the effort though.

Reply:
Thought about returning an error and throwing an exception in that case. But actually when updating the plugin fails and some plugin would be incompatible then, they would be disabled afterwards. Not sure if we should report that as failed update 🤔

Reply:
hmm, maybe not